src/Libraries/RevitNodes/Elements/DirectShape.cs

```diff
@@ -0,0 +1,403 @@
+        /// The category of a DirectShape cannot be changed after placing a DirectShape, so
+        /// a new DirectShape will be generated if the category input is changed
+        /// </summary>
+        /// <param name="geometry">a solid that will be tessellated and placed in the Revit model as a DirectShape</param>
+        /// <param name="name">a string name for the directshape</param>
+        /// <param name="category">must be a top level built-in category</param>
+        /// <param name="material">a material to apply to the faces of the DirectShape</param>
+        /// <returns>a DirectShape Element</returns>
+        public static DirectShape ByGeometryNameCategoryMaterial(Autodesk.DesignScript.Geometry.Geometry geometry, string name, Category category,Material material)
+        {
+
+            if (geometry is Autodesk.DesignScript.Geometry.Solid || geometry is Autodesk.DesignScript.Geometry.Surface)
+            {
+                return new DirectShape(geometry, name, new ElementId(category.Id), new ElementId(material.Id));
+            }
+
+            throw new ArgumentException(Revit.Properties.Resources.DirectShapeInvalidArgument);
+
+        }
+
+        /// <summary>
+        /// Create a Revit DirectShape given some geometry, a name for the shape, category, and material.
+        /// The geometry will be tessellated before being placed in the Revit model
+        /// The category of a DirectShape cannot be changed after placing a DirectShape, so
+        /// a new DirectShape will be generated if the category input is changed
+        /// </summary>
+        /// <param name="mesh">a mesh that will be placed in the Revit model as a DirectShape</param>
+        /// <param name="name">a string name for the directshape</param>
+        /// <param name="category">must be a top level built-in category</param>
+        /// <param name="material">a material to apply to the faces of the DirectShape</param>
+        /// <returns>a DirectShape Element</returns>
+        public static DirectShape ByMeshNameCategoryMaterial(Autodesk.DesignScript.Geometry.Mesh mesh, string name, Category category, Material material)
+        {
+            return new DirectShape(mesh, name, new ElementId(category.Id), new ElementId(material.Id));
+        }
+
+        #endregion
+
+        #region Internal static constructors
+
+        /// <summary>
+        /// Construct a DirectShape from the Revit document. 
+        /// </summary>
+        /// <param name="directShape"></param>
+        /// <param name="isRevitOwned"></param>
+        /// <returns></returns>
+        internal static DirectShape FromExisting(Autodesk.Revit.DB.DirectShape directShape, bool isRevitOwned)
+        {
+            return new DirectShape(directShape)
+            {
+                IsRevitOwned = isRevitOwned
+            };
+        }
+
+        #endregion
+
+        public override string ToString()
+        {
+            return InternalDirectShape.Name;
```

Comment:
I'm guessing this produces something sensible when the user has not supplied a name?